src/artm/regularizer_interface.h

```diff
@@ -36,7 +37,8 @@ class RegularizerInterface {
                                const ModelConfig& model_config,
                                int inner_iter,
                                double tau,
-                               ::artm::utility::DenseMatrix<float>* theta) { return true; }
+                               ::artm::utility::DenseMatrix<float>* theta,
+                               std::shared_ptr<const artm::core::TopicModel> topic_model) { return true; }
```

Comment:
std::shared_ptr<const artm::core::TopicModel>  => const TopicModel&
You API may cause some users to do this ugly thing:
http://stackoverflow.com/questions/12127758/pass-an-object-to-a-function-expecting-shared-ptr-without-actually-sharing-owner